ddprof-exporter/src/connector/mod.rs

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use futures::future::BoxFuture;
+use futures::{future, FutureExt};
+use hyper::client::HttpConnector;
+use rustls::ClientConfig;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+#[cfg(unix)]
+pub mod uds;
+
+mod conn_stream;
+use conn_stream::{ConnStream, ConnStreamError};
+
+#[derive(Clone)]
+pub enum MaybeHttpsConnector {
```

Comment:
Slight nitpick, because the connector also handles UDS even if it's not a member of the enum because we don't need to have any associated state, `MaybeHttpsConnector` is IMO not a great name.

Reply:
>MaybeHttpsConnector is IMO not a great name.
😅 
I was "inspired" by one of underlying classes -> MaybeHtttpsStream :)

I think I can rename it back now to just Connector (I used MaybeHttpsConnector name, when working on refactoring while I had old Connector still present.